### PR TITLE
Add Typescript annotations to TableSelectRow

### LINKED ISSUE #12531: Add TypeScript types to TableSelectRow

### PR DESCRIPTION
**Pull request by jpsorensen**
Closes #12531 

Convert existing propType definitions to Typescript annotations on the TableSelectRow component. This is part of a broader effort to add Typescript annotations to components, tracked in https://github.com/carbon-design-system/carbon/issues/12513.

#### Changelog

**Changed**

- TableSelectRow.js has been converted to tsx

#### Testing / Reviewing

- In the packages/react directory, run yarn build, yarn storybook. In the storybook, navigate to DataTable>Selection and confirm that the component's behavior remains unchanged.
- Past usages of TableSelectRow at the bottom of TableSelectRow.tsx. Confirm that they compile as expected. Modify inputs so that they have the wrong types and confirm that it does not compile. Remove a required parameter and confirm that it does not compile. (I used the examples from the TableSelectRow unit tests).

### CODE REVIEW

File: packages/react/src/components/DataTable/TableSelectRow.tsx [hunk 1]
```diff
@@ -12,6 +12,57 @@ import InlineCheckbox from '../InlineCheckbox';
...
 import { usePrefix } from '../../internal/usePrefix';
 
+export interface TableSelectRowProps {
+  /**
+   * Specify the aria label for the underlying input control
+   */
+  ariaLabel: string;
+
+  /**
+   * Specify whether all items are selected, or not
```
Comment (jpsorensen) [inline packages/react/src/components/DataTable/TableSelectRow.tsx]: I'm guessing that this was accidentally copied over from TableSelectAll. I decided to leave it as-is (leave it in the PropTypes docs, and to have the Typescript docs mirror the PropTypes docs) in this PR so as not to get a separate change mixed up in this Typescript conversion. Just let me know if you'd prefer that I fix this as part of this PR!

File: packages/react/src/components/DataTable/TableSelectRow.tsx [hunk 2]
```diff
@@ -34,7 +85,8 @@ const TableSelectRow = ({
   };
   const InlineInputComponent = radio ? RadioButton : InlineCheckbox;
   const tableSelectRowClasses = classNames(`${prefix}--table-column-checkbox`, {
-    [className]: className,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    [className!]: className,
```
Comment (francinelucca) [inline packages/react/src/components/DataTable/TableSelectRow.tsx]: think this way you can forego having to disable the eslint rule
```suggestion
    ...(className && {[className]: true}),
```

File: packages/react/src/components/DataTable/TableSelectRow.tsx (hunk 1 above)
Comment (francinelucca) [reply packages/react/src/components/DataTable/TableSelectRow.tsx]: nice catch! I do believe that's a mistake, I think it'll be fine to fix here as to not carry over more mistakes, we can rename here and in the proptypes to something like "Specify whether row is selected, or not" if you don't mind adding it 🙏🏻

**Review by francinelucca — APPROVED** [review]
Thanks for getting this PR up! LGTM, just two minor details

File: packages/react/src/components/DataTable/TableSelectRow.tsx (hunk 2 above)
Comment (jpsorensen) [reply packages/react/src/components/DataTable/TableSelectRow.tsx]: Done, thank you!

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**